yogstation/code/game/gamemodes/traitor/double_agents.dm

```diff
@@ -0,0 +1,47 @@
+/datum/game_mode/traitor/internal_affairs/make_antag_chance()
+	return FALSE
+
+/datum/antagonist/traitor/internal_affairs/iaa_process()
+	return
+
+/datum/game_mode/traitor/internal_affairs/check_finished(force_ending) //to be called by SSticker
+	.=..()
+	if(force_ending || !SSticker.setup_done || !gamemode_ready || .) //if the gamemode is already considered finished, if it's not begun or if it's being force ended
+		return .
+
+	var/new_target
+	for(var/datum/mind/traitor in traitors)
+		if(!traitor.current || traitor.current.stat==DEAD)
+			continue
+		new_target = TRUE
+		for(var/datum/objective/O in traitor)
+			if(O.check_completion() || istype(O, /datum/objective/escape) || istype(O, /datum/objective/survive))
+				continue
+			new_target = FALSE
+
+		if(new_target)
+			var/list/other_traitors = traitors
```

Comment:
Can't you do ``var/list/other_traitors = traitors - traitor`` here and delete the ``other_traitor == traitor`` check a few lines below this?